### PR TITLE
Popup error

### PR DESCRIPTION
**Pull request by NafisFaisalKhan**
@andrew-ironforge

### CODE REVIEW

File: src/components/widgets/RaceTrackerMap/RaceTrackerMap.tsx [hunk 1]
```diff
@@ -126,7 +130,10 @@ class RaceTrackerMap extends React.Component<IRaceTrackerMapProps, any> {
         mapboxApiAccessToken={TOKEN}
         mapStyle={this.state.mapStyle}
         {...this.state.viewport}
-        onViewportChange={this.__handleViewportChange}>
+        onViewportChange={this.__handleViewportChange}
+        onClick={this._onClick}>
```
Comment (andrew-ironforge) [inline src/components/widgets/RaceTrackerMap/RaceTrackerMap.tsx]: `this.__handleClick`

File: src/components/widgets/RaceTrackerMap/RaceTrackerMap.tsx [hunk 2]
```diff
@@ -126,7 +130,10 @@ class RaceTrackerMap extends React.Component<IRaceTrackerMapProps, any> {
...
+        
+         {this._renderPopup()}
```
Comment (andrew-ironforge) [inline src/components/widgets/RaceTrackerMap/RaceTrackerMap.tsx]: convention for `private` functions is 2 _ in front of the name:
`this._renderPopup()` -> `this.__renderPopup()`

File: src/components/widgets/RaceTrackerMap/RaceTrackerMap.tsx [hunk 3]
```diff
@@ -84,6 +85,9 @@ class RaceTrackerMap extends React.Component<IRaceTrackerMapProps, any> {
               ]
             },
             properties: {
+              description: `<h3>Racer ID: `+trackLeader['trackleaders_racer_ID'][0]+`</h3>
```
Comment (andrew-ironforge) [inline src/components/widgets/RaceTrackerMap/RaceTrackerMap.tsx]: If you're going to use the backtick ( ` ) to do template strings, I would recommend that you make use of the string interpolation. This block of code becomes
```
properties: {
  description: `
    <h3>
      Racer ID: ${trackLeader['trackleaders_racer_ID'][0]}
    </h3> 
    <strong>
      Device Battery State: ${trackLeader.message[0]['batteryState'][0]}
    </strong>
  `,
  speed: 10
}
```

This way, it's a lot more clear what the structure is of what you're trying to do, and it's more readable.

Comment (andrew-ironforge) [reply src/components/widgets/RaceTrackerMap/RaceTrackerMap.tsx]: The issue with your popup is the fact that you're passing a string containing HTML. `{popup.description}` only sees a string, not the fact that it needs to render HTML. When you click on a marker on the map, you need to `setState` the info you need. And inside of `<Popup></Popup>` you need to write out the HTML structure with JSX that you are currently putting in `description`.

File: src/components/widgets/RaceTrackerMap/RaceTrackerMap.tsx [hunk 4]
```diff
@@ -140,6 +147,41 @@ class RaceTrackerMap extends React.Component<IRaceTrackerMapProps, any> {
...
+                lng: event.lngLat[0],
+                description:feature.properties.description
+            } 
+          })
+         
+       }
+    }
+  };
+
+ private  _renderPopup() {
```
Comment (andrew-ironforge) [inline src/components/widgets/RaceTrackerMap/RaceTrackerMap.tsx]: Referencing my above comment about writing out the HTML, your function would become: 
```
private _renderPopup() {
    const {popup} = this.state;
    return (
      popup && (
        <Popup
          tipSize={5}
          anchor="top"
          longitude={popup.lng}
          latitude={popup.lat}
          closeOnClick={false}
          onClose={() => this.setState({popup: null})}>
          <div>
            <h3>
              Racer ID: {this.state.selectedTrackLeaderId}
            </h3>
            <strong>
              Device Battery State: {this.state.selectedTrackLeaderBatteryState}
            </strong>
          </div>
        </Popup>
      )
    );
  }
```

Comment (andrew-ironforge) [reply src/components/widgets/RaceTrackerMap/RaceTrackerMap.tsx]: In this example, I assume you've `setState` the data you need, and then reference here.

Comment (andrew-ironforge) [reply src/components/widgets/RaceTrackerMap/RaceTrackerMap.tsx]: Additionally, `_renderPopup` should become `__renderPopup`, and should also be an arrow function:
`private __renderPopup = () => {};`

**Comment by NafisFaisalKhan** [comment]
@andrew-ironforge I added pagination to the query.

**Comment by andrew-ironforge** [comment]
Closed the PR, as I was unsure what the status was. @NafisFaisalKhan when you are ready to have your pagination and such merged, please open another PR

**Pull request closed without merging**